crypt-port.h

```diff
@@ -68,6 +68,81 @@
 #define MIN_SIZE(x) (x)
 #endif
 
+/* Macros for detecting endianness of the system at compile time.  */
```

Comment:
Hm.  Where did you get this big list of macros?  How do we know it's exhaustive? Is there any way we could plausibly make it shorter?  Two possiblities come to mind: make use of `sys/endian.h` where available, and keep the autoconf check around and limit the cases in the if statement to the ones that are relevant when `IS_BIGENDIAN` is undefined.

Reply:
I've re-added and modified the configure check in the rebased commit.  The macros are used only when configure sets endianness to `universal`.

BTW, most of the macros are comming from https://sourceforge.net/p/predef/wiki/Endianness/ and the variations of underscores were added by me, since some platforms define those macros in different variations.